contents/tutorials/go-analytics.md

```diff
@@ -0,0 +1,300 @@
+                e.preventDefault();
+                $.ajax({
+                    url: $(this).attr('action'),
+                    type: 'POST',
+                    data: $(this).serialize(),
+                });
+            });
+        });
+    </script>
+</body>
+</html>
+```
+
+Next, we install the [`gorilla/sessions`](https://github.com/gorilla/sessions) package to manage our sessions:
+
+```bash
+go get github.com/gorilla/sessions
+```
+
+Lastly, we set up `main.go` with all our route handling:
+
+```go file=main.go
+package main
+
+import (
+    "html/template"
+    "net/http"
+    "github.com/gorilla/sessions"
+)
+
+var (
+    store = sessions.NewCookieStore([]byte("your-secret-key"))
+    templates = template.Must(template.ParseGlob("*.html"))
+)
+
+func main() {
+    http.HandleFunc("/", loginHandler)
+    http.HandleFunc("/auth", authHandler)
+    http.HandleFunc("/dashboard", dashboardHandler)
+    http.HandleFunc("/api/dashboard", apiDashboardHandler)
+
+    http.ListenAndServe(":8000", nil)
+}
+
+func loginHandler(w http.ResponseWriter, r *http.Request) {
+    templates.ExecuteTemplate(w, "login.html", nil)
+}
+
+func authHandler(w http.ResponseWriter, r *http.Request) {
+    session, _ := store.Get(r, "session-name")
+
+    session.Values["email"] = r.FormValue("email")
+    session.Values["name"] = r.FormValue("name")
+    session.Values["company_name"] = r.FormValue("company_name")
+    session.Save(r, w)
+
+    http.Redirect(w, r, "/dashboard", http.StatusFound)
+}
+
+func dashboardHandler(w http.ResponseWriter, r *http.Request) {
+    session, _ := store.Get(r, "session-name")
+    data := map[string]interface{}{
+        "Email": session.Values["email"],
+        "Name": session.Values["name"],
+        "CompanyName": session.Values["company_name"],
+    }
+    templates.ExecuteTemplate(w, "dashboard.html", data)
+}
+
+func apiDashboardHandler(w http.ResponseWriter, r *http.Request) {
+    // This API is called when the button is pressed
+    // We'll add code later in the tutorial
+}
+```
+
+Run `go run main.go` and navigate to `http://localhost:8000` to see our app in action. Enter anything in the `log-in` page to save some session details.
+
+![Basic Go app](../images/tutorials/go-analytics/basic-go-app.mp4)
+
+## 2. Add PostHog to your app
+
+With our app set up, it’s time to install and set up PostHog. If you don't have a PostHog instance, you can [sign up for free](https://us.posthog.com/signup).
+
+Run `go get github.com/posthog/posthog-go` to add [PostHog's Go SDK](/docs/libraries/php) as a dependency. Then, initialize PostHog in the `main()` method in `main.go`.
+
+To do this, you need your project API key and instance address from [your project settings](https://us.posthog.com/project/settings). You also need to [create a personal API key](https://us.posthog.com/settings/user-api-keys). Use these values to initialize your client using `posthog.NewWithConfig()`: 
+
+```go file=main.go
```

Comment:
github.com/posthog/posthog-go needs to be added to imports?

Reply:
I think my IDE added this automatically, but will add it